### PR TITLE
feat!: remove immutable

### PR DESCRIPTION
**Pull request by pyphilia**

### CODE REVIEW

File: src/components/item/settings/ThumbnailSetting.tsx [hunk 1]
```diff
@@ -143,9 +142,8 @@ const ThumbnailSetting = ({ item }: Props): JSX.Element | null => {
           // TODO: fix type
           url={
             imgUrl ??
-            (item as EmbeddedLinkItemTypeRecord)?.extra?.[
-              ItemType.LINK
-            ]?.thumbnails?.first()
+            (item as EmbeddedLinkItemType)?.extra?.[ItemType.LINK]
```
Comment (spaenleh) [inline src/components/item/settings/ThumbnailSetting.tsx:145]: First check the type and then the type should be inferred correctly.
```suggestion
            (item.type === ItemType.LINK ? item.extra.[ItemType.LINK]
```

File: src/components/layout/Navigation.tsx [hunk 2]
```diff
@@ -46,7 +46,8 @@ const Navigator = (): JSX.Element | null => {
   });
 
   const isParentOwned =
-    (item?.creator?.id ?? parents?.first()?.creator?.id) === currentMember?.id;
+    currentMember &&
```
Comment (spaenleh) [inline src/components/layout/Navigation.tsx:49]: Is this the breadcrumb fix ?

File: src/components/main/ItemScreen.tsx [hunk 3]
```diff
@@ -116,6 +116,7 @@ const WrappedItemScreen = (): JSX.Element => {
   const Component = ItemLoginAuthorization({
     signIn: itemLoginSignIn,
     itemId,
+    // TODO: do not use empty user for current member
```
Comment (spaenleh) [inline src/components/main/ItemScreen.tsx:119]: Is this fixed already ?

File: src/utils/itemExtra.ts
Comment (spaenleh) [inline src/utils/itemExtra.ts:1]: I do not understand these changes ? Was it not versatile enough ?

**Review by spaenleh — APPROVED** [review]
🛥️ Woop Woop 🔥

File: src/components/item/form/FileForm.tsx [hunk 4]
```diff
@@ -28,7 +28,9 @@ const FileForm = (props: EditModalContentPropType): JSX.Element | null => {
   ) {
     const itemExtra = getExtraFromPartial(item);
     const { mimetype, altText } = itemExtra;
-    const updatedExtra = getExtraFromPartial(updatedProperties);
+    const updatedExtra = updatedProperties.extra
+      ? getExtraFromPartial(updatedProperties)
+      : {};
```
[the remark this answers is not in the document]
Comment (pyphilia) [reply src/components/item/form/FileForm.tsx:33]: `getExtraFromPartial` cannot be undefined..

File: src/components/layout/Navigation.tsx (hunk 2 above)
Comment (pyphilia) [reply src/components/layout/Navigation.tsx:49]: I guess? At least it makes sense I think.

**Pull request merged**